enterprise/backend/src/metabase_enterprise/gsheets/api.clj

```diff
@@ -296,15 +304,30 @@
                           :last_sync_at (.getEpochSecond ^Instant (t/instant last-sync-at))
                           :hm/response (loggable-response hm-response))
             (analytics/inc! :metabase-gsheets/connection-creation-error {:reason "status_error"})
-            (log/errorf "Error getting status of connection %s: %s %s" conn-id (:status-reason hm-body) (:error-detail hm-body)))))
-      (if (empty? saved-setting)
+            (log/errorf "Error getting status of connection %s: status-reason=`%s` error-detail=`%s`" conn-id (:status-reason hm-body) (:error-detail hm-body)))))
+      (cond
+        (empty? saved-setting)
         {:status "not-connected"}
+
+        (= 403 hm-status-code)
+        (let [[list-status list-response] (hm-get-all-connections)]
+          (if (and (= :ok list-status) (not (some (partial = conn-id) (set (map :id (:body list-response))))))
+            (u/prog1 {:status "not-connected"}
+              (log/errorf "Removing google drive connection %s because harbormaster gives a 403 error for it." conn-id)
```

Comment:
Maybe WARN or INFO level? Always think of error-level as for logging exceptions.